File: in_Python/0409 Longest Palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        res = 0
        dict = {}
        for x in list(s):
            if x not in dict:
                dict[x] = 1
            else:
                dict[x] += 1
        max_odd = 0
        flag = False
        for k in dict:
            if dict[k] % 2 == 0:
                res += dict[k]
            else:
                flag = True
                res += dict[k]-1
        if flag: res += 1
        return res
        

File: in_Java/0409 Longest Palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        HashMap<Character,Integer> hm = new HashMap<>();
        for(char c : s.toCharArray()){
            if(!hm.containsKey(c)){
                hm.put(c,1);
            }else{
                hm.put(c,hm.get(c)+1);
            }
        }
        int res = 0;
        boolean flag = false;
        for(int i : hm.values()){
            if(i % 2 == 0) res += i;
            else{
                flag = true;
                res += i-1;
            }            
        }
        return flag? res+1 : res;
    }
}